docs/upgrade-guides/3-19-to-3-20.mdx

```diff
@@ -13,4 +13,151 @@ Otherwise, you will need to downtime your solution to ensure correct data migrat
+
+:::info
+By "paid" in this guide we mean that the `TransactionItems` connected to `Checkout` have fully covered [`Checkout.totalPrice`](/api-reference/checkout/objects/checkout.mdx#checkouttotalpricetaxedmoney---), which means that [`Checkout.authorizeStatus`](/api-reference/checkout/objects/checkout.mdx#checkoutauthorizestatuscheckoutauthorizestatusenum---) is set to [`FULL`](/api-reference/checkout/enums/checkout-authorize-status-enum.mdx).
+
+Learn more about this in [Transaction docs](/developer/payments/transactions.mdx#automatic-checkout-completion)
+:::
+
+This feature is **disabled** after migration, and needs to be enabled manually in the [channel settings](/developer/channels/configuration.mdx#settings).
+
+New Saleor instances created from version 3.20 onwards will have this feature **enabled by default**.
+
+This removes the need for storefront implementation to call `checkoutComplete` mutation in order to create `Order`.
+
+Previously a store might have had missing orders, due to `checkoutComplete` mutation not being always called. This could have happened for many unexpected reasons: network issues, storefront bug, customer leaving website, closing tab before being redirected from payment provider, etc.
+
+:::tip
+Before Saleor 3.20 you might have used other measures to mitigate missing orders issue, for example by using `CHECKOUT_FULLY_PAID` webhook and calling `checkoutComplete` mutation on your backend.
+
+After migrating to Saleor 3.20 and enabling this feature for all your channels, you can safely get rid of this workaround.
+
+Keep in mind that `CHECKOUT_FULLY_PAID` webhook would have only been triggered if you also used `TransactionFlowStrategy.CHARGE` for your channel. Automatic checkout completion doesn't have this problem.
+:::
+
+
+### Before you enable this feature
+
+
+:::danger
+Enabling this feature might break your current storefront implementation and services connected to Saleor that use `Checkout` object.
+:::
+
+Before you enable this feature make sure that you don't rely on any `Checkout` query or mutation that updates `Checkout` after transaction is created.
+
+:::tip
+After calling `transactionInitialize` or `transactionProcess` you can determine if `Checkout` has been completed by checking the `TransactionEvent` object returned in the mutation response.
+
+Transaction is considered "paid" when it has one of the following `types`:
+
+- `AUTHORIZATION_SUCCESS`
+- `AUTHORIZATION_REQUEST`
+- `CHARGE_SUCCESS`
+- `CHARGE_REQUEST`
+
+and it's `amount` is equal to `Checkout.totalPrice`.
+
+Note that payment provider might return **lower** amount than the one you have requested (in that case `Checkout` is not paid). Your logic also needs to take into consideration multiple Transactions if you use split-payments.
+:::
+
+:::tip
+You can also determine if `Checkout` has been paid by calling `checkoutComplete` mutation.
```

Comment:
In this situation we **won't** have `Checkout` object, after making payment automatic checkout completion will kick-in and convert it into an `Order`.

This is why we can't use these fields. I agree that this shouldn't be the **default** way of checking payment status, if you think this is misleading then I can think of some way to re-word that tip. @korycins 

Maybe we could rethink if `checkout` query should work some time after `Checkout` is converted into an order, so that storefront could fetch the `authorizeStatus`?

Reply:
If this is a blocking comment how about @korycins I remove it from this PR and create a separate one?

Reply:
@witoszekdev @korycins maybe we should add option to fetch order by checkout token?

Reply:
Just re-phrase the section, and emphasize that in case of missing checkout, calling checkoutComplete would provide you an order. 
We want to be sure, that we're clear here, that the basic step of defining the payment status is `checkout.authorizeStatus`.

Reply:
We don't want to have a situation, where someone is hitting checkoutComple for the whole time, to check if order is actually paid